packages/devtools_app/lib/src/ui/hover.dart

```diff
@@ -447,16 +446,20 @@ class _HoverCardTooltipState extends State<HoverCardTooltip> {
     if (_currentHoverCard != null) {
       // If the widget that triggered the hovercard is disposed, then the
       // HoverCard should be removed from the screen
-      _hoverCardTooltipController.removeHoverCard(_currentHoverCard!);
+      _hoverCardController.removeHoverCard(_currentHoverCard!);
     }
     widget.disposable?.dispose();
     super.dispose();
   }
 
+  @override
+  void didChangeDependencies() {
+    _hoverCardController = Provider.of<HoverCardController>(context);
+    super.didChangeDependencies();
```

Comment:
call to super first

Reply:
ouu interesting.
<img width="484" alt="Screen Shot 2022-09-22 at 1 27 48 PM" src="https://user-images.githubusercontent.com/1386322/191812861-2037bff1-723e-4c30-b02b-bf1b1d505e46.png">

The default template puts the TODO at the top, so I had put it above the super purposely.

Reply:
@DanTup this would be a good polish item for VS code. To add the TODO where the implementation is supposed to go.

Reply:
Yeah, seems like this should probably be the other way around. Although, there's also an issue open about removing TODOs at https://github.com/dart-lang/sdk/issues/49331. I'll ping on there and see if there are opinions of changing this order vs just removing (the change will be in the Dart analysis server rather than VS Code, as it handles most code edits/generation like this).